src/components/ResourceList/tests/ResourceList.test.tsx

```diff
@@ -750,6 +753,118 @@ describe('<ResourceList />', () => {
+            )
+            .find('input[type="checkbox"]');
+
+          expect(deselectAllCheckbox.instance()).toBe(document.activeElement);
+        });
+
+        it('focuses the plain CheckableButton checkbox when items are selected and the deselect Checkable button the is clicked', () => {
+          const resourceList = mountWithAppProvider(
+            <ResourceList
+              items={itemsWithID}
+              renderItem={renderItem}
+              selectedItems={allSelectedIDs}
+              promotedBulkActions={promotedBulkActions}
+            />,
+          );
+
+          const deselectAllCheckableButton = resourceList.findWhere(
+            (wrap) => wrap.is(CheckableButton) && !wrap.prop('plain'),
+          );
+
+          trigger(deselectAllCheckableButton, 'onToggleAll');
+
+          const selectAllCheckableCheckbox = resourceList
+            .findWhere(
+              (wrap) => wrap.is(CheckableButton) && wrap.prop('plain') === true,
+            )
+            .find('input[type="checkbox"]');
+
+          expect(selectAllCheckableCheckbox.instance()).toBe(
+            document.activeElement,
+          );
+        });
+      });
+
+      describe('small screen', () => {
+        afterEach(() => {
+          setDefaultScreen();
+        });
+
+        it('keeps focus on the CheckableButton checkbox when selecting', () => {
+          setSmallScreen();
+
+          const resourceList = mountWithAppProvider(
+            <ResourceList
+              items={itemsWithID}
+              renderItem={renderItem}
+              promotedBulkActions={promotedBulkActions}
+            />,
+          );
+
+          trigger(resourceList.find(Button), 'onClick');
+
+          const selectAllCheckableButton = resourceList.findWhere(
+            (wrap) => wrap.is(CheckableButton) && !wrap.prop('plain'),
+          );
+
+          trigger(selectAllCheckableButton, 'onToggleAll');
+
+          const checkBox = selectAllCheckableButton.find(
+            'input[type="checkbox"]',
+          );
+
+          expect(checkBox.instance()).toBe(document.activeElement);
+        });
+
+        it('keeps focus on the CheckableButton checkbox when deselecting', () => {
+          setSmallScreen();
+
+          const resourceList = mountWithAppProvider(
+            <ResourceList
+              items={itemsWithID}
+              selectedItems={allSelectedIDs}
+              renderItem={renderItem}
+              promotedBulkActions={promotedBulkActions}
+            />,
+          );
+
+          const deselectAllCheckableButton = resourceList.findWhere(
+            (wrap) => wrap.is(CheckableButton) && !wrap.prop('plain'),
+          );
+
+          trigger(deselectAllCheckableButton, 'onToggleAll');
+
+          const checkBox = deselectAllCheckableButton.find(
+            'input[type="checkbox"]',
+          );
+
+          expect(checkBox.instance()).toBe(document.activeElement);
```

Comment:
`instance`  returns a react component, however, `getDOMNode` returns the dom element

Reply:
Interesting how this did pass though.